### ISSUE TITLE
String Formatting

### ISSUE THREAD

**Issue by animaone** [pyscript/pyscript#173]
I think string formatting will be a very big issue for pyscript in the next months/years. Maybe a generic solution similar to `<pre> </pre>` could be possible? I've seen a similar solution in a syntax highlighting library some time ago (`<code> </code>`). For now, the following code returns an error(in both Firefox and Chrome):

```python
for r in range(0xff):
	color="#%02x29%02x" % (r,r)
	printdata = "<font color=\"%s\">sodjf</font>" % color
	print(printdata)
```

JsException(PythonError: Traceback (most recent call last):
  File "/lib/python3.10/site-packages/_pyodide/_base.py", line 429, in eval_code
    .run(globals, locals)
  File "/lib/python3.10/site-packages/_pyodide/_base.py", line 300, in run
    coroutine = eval(self.code, globals, locals)
  File "", line 5, in 
TypeError: not all arguments converted during string formatting
)

**Comment by verhulstm** [comment]
can you test it out with f-Strings?

**Comment by animaone** [comment]
f-Strings don't return an error but the print command is not printing everything.

```python
print('<font color="red">should be red</font>')

for r in range(0xff):
	color="#%02x29%02x" % (r,r)
	printdata = "<font color=\"{color}\">should be some color</font>"
	print(printdata)

```

it prints <div>should be red</div>

**Comment by animaone** [comment]
same 

```
print('<font color="red">should be red</font>')

for r in range(0xff):
	color=f"#{r:#02x}29{r:#02x}"
	printdata = f"<font color=\"{color}\">should be some color</font>"
	print(printdata)
```

**Comment by verhulstm** [comment]
interesting. let me look more into this. thanks for finding this

**Comment by JeffersGlass** [comment]
I believe this is fixed by #684 - at least, the final example posted above does correctly work. 

Conversations around escaping HTML-relevant characters in PyScript source are ongoing in #735 - but this specific issue is resolved.
